File: .claude/api docs/Pazaryeri ve Direkt API/PayTR Platform Transfer Talebi/Türkçe/Örnek PYTHON/transfer_sonucu_ornek.py
# Python 3.6+
# Django Web Framework referans alınarak hazırlanmıştır

import base64
import hashlib
import hmac
import json

from django.shortcuts import render, HttpResponse
from django.views.decorators.csrf import csrf_exempt

@csrf_exempt
def callback(request):
    if request.method != 'POST':
        return HttpResponse(str(''))

    post = request.POST

    # API Entegrasyon Bilgileri - Mağaza paneline giriş yaparak BİLGİ sayfasından alabilirsiniz.
    merchant_key = b'YYYYYYYYYYYYYY'
    merchant_salt = 'ZZZZZZZZZZZZZZ'

    # Bu kısımda herhangi bir değişiklik yapmanıza gerek yoktur.
    # POST değerleri ile hash oluştur.
    post['trans_ids'] = post['trans_ids'].replace('\\', '')
    hash_str = post['trans_ids'] + merchant_salt
    hash = base64.b64encode(hmac.new(merchant_key, hash_str.encode(), hashlib.sha256).digest())

    # Oluşturulan hash'i, paytr'dan gelen post içindeki hash ile karşılaştır
    # (isteğin paytr'dan geldiğine ve değişmediğine emin olmak için)
    # Bu işlemi güvenlik nedeniyle mutlaka yapmanız gerekiyor.
    if hash != post['hash']:
        return HttpResponse(str('PAYTR notification failed: bad hash'))

    # post['trans_ids'] içerisinde daha önce PayTR'a ilettiğiniz transfer taleplerinden tamamlanan transferlerin trans_id bilgileri JSON formatında gelir
    # trans_id bilgisi transfer talebi yaparken PayTR'a gönderdiğiniz her işlem için eşsiz değerdir
    trans_ids = json.loads(post['trans_ids'])

    for ids in trans_ids:
        # Örn: Burada trans_id ile veritabanınızdan transfer talebini tespit edip ilgili kullanıcınıza bilgilendirme gönderebilirsiniz (email, sms vb.)
        print(ids)

    # Bildirimin alındığını PayTR sistemine bildir.
    return HttpResponse(str('OK'))

File: .claude/api docs/Pazaryeri ve Direkt API/PayTR Platform Transfer Talebi/Türkçe/Örnek PHP/transfer_sonucu_ornek.php
<?php

	$post = $_POST;

	####################### DÜZENLEMESİ ZORUNLU ALANLAR #######################
	#
	## API Entegrasyon Bilgileri - Mağaza paneline giriş yaparak BİLGİ sayfasından alabilirsiniz.
	$merchant_key 	= 'YYYYYYYYYYYYYY';
	$merchant_salt	= 'ZZZZZZZZZZZZZZ';
	###########################################################################

	####### Bu kısımda herhangi bir değişiklik yapmanıza gerek yoktur. #######
	#
	## POST değerleri ile hash oluştur.
    $post["trans_ids"]=str_replace("\\", "", $post["trans_ids"]);
	$hash = base64_encode( hash_hmac('sha256', $post['trans_ids'].$merchant_salt, $merchant_key, true) );
	#
	## Oluşturulan hash'i, PayTR'dan gelen post içindeki hash ile karşılaştır (isteğin PayTR'dan geldiğine ve değişmediğine emin olmak için)
	## Bu işlemi güvenlik nedeniyle mutlaka yapmanız gerekiyor.
	if( $hash != $post['hash'] )
		die('PAYTR notification failed: bad hash');
	###########################################################################

	## $post['trans_ids'] içerisinde daha önce PayTR'a ilettiğiniz transfer taleplerinden tamamlanan transferlerin trans_id bilgileri JSON formatında gelir
	## trans_id bilgisi transfer talebi yaparken PayTR'a gönderdiğiniz her işlem için eşsiz değerdir
	$trans_ids = json_decode($post['trans_ids'],1);
	foreach($trans_ids as $trans_id)
	{
		## Örn: Burada $trans_id ile veritabanınızdan transfer talebini tespit edip ilgili kullanıcınıza bilgilendirme gönderebilirsiniz (email, sms vb.)
	}

	## Bildirimin alındığını PayTR sistemine bildir.
	echo "OK";
	exit;
?>